logpyle/__init__.py

```diff
@@ -287,14 +287,9 @@ def _get_unique_id():
         return uuid1().hex
 
 
-def _get_random_suffix(n):
-    characters = (
-            [chr(65+i) for i in range(26)]
-            + [chr(97+i) for i in range(26)]
-            + [chr(48+i) for i in range(10)])
-
-    from random import choice
-    return "".join(choice(characters) for i in range(n))
+def _get_unique_suffix():
+    from datetime import datetime
+    return "-" + datetime.utcnow().strftime("%Y%m%d%H%M%S")
```

Comment:
This is actually unlikely to be sufficiently unique. Could you put some of those random characters back?

Reply:
Instead of adding random characters (which would again break item 2 of my list above), could we just add millisecond resolution to the timer? Conflicts should already be handled by the existing code (but might delay the logging startup a bit).

Reply:
> Conflicts should already be handled by the existing code (but might delay the logging startup a bit).

How so?

>  (which would again break item 2 of my list above)

Why? If it's timestamp+random characters?

Reply:
> > Conflicts should already be handled by the existing code (but might delay the logging startup a bit).
> 
> How so?

Through the 'retry' mechanism in https://github.com/illinois-ceesd/logpyle/blob/cb7bfec1ab793a1ff1fd0be76c97b6b7135bb719/logpyle/__init__.py#L493

> > (which would again break item 2 of my list above)
> 
> Why? If it's timestamp+random characters?

Because files with the same timestamp (but different random characters) would then be ordered randomly. One goal of this patch is to make it easier to answer the question "Which file belongs to my most recent run?"

Reply:
> would then be ordered randomly

I don't see this. If there's a leading timestamp and the suffix matches for all ranks, then the sorting should be the same as otherwise?

> Through the 'retry' mechanism

OIC

Reply:
> > would then be ordered randomly
> 
> I don't see this. If there's a leading timestamp and the suffix matches for all ranks, then the sorting should be the same as otherwise?

I think the relevant case is when a conflict could happen in the current code: two separate logging instances starting very close to each other:
- in the current version of this PR, this would engage the retry mechanism, leading to a startup delay of up to 1 second for the later logging instance.
  - hence my suggestion of using milliseconds instead, which would all but eliminate this issue
- in your suggestion, there would be no delay, but there would be no way of saying which instance was earlier based on the file name:
   ```
   mpi-log-rank0-20201202203827-abcd.sqlite
   mpi-log-rank0-20201202203827-efgh.sqlite
   ```

Reply:
Fair enough.